ufs2arco/regrid_mom6.py

```diff
@@ -0,0 +1,302 @@
+
+        # open input dataset
+        ds_in_t = ds_in.rename({"xh": "lon", "yh": "lat"})
+        ds_in_u = ds_in.rename({"xq": "lon", "yh": "lat"})
+        ds_in_v = ds_in.rename({"xh": "lon", "yq": "lat"})
+
+        # open rotation dataset
+        ds_rot = xr.open_dataset(self.rotation_file)
+        ds_rot = ds_rot[["cos_rot", "sin_rot"]]
+        self.ds_rot = ds_rot.rename({"xh": "lon", "yh": "lat"})
+
+        # create output dataset
+        lons, lats = np.meshgrid(self.lons1d_out, self.lats1d_out)
+        grid_out = xr.Dataset()
+        grid_out["lon"] = xr.DataArray(lons, dims=["nx", "ny"])
+        grid_out["lat"] = xr.DataArray(lats, dims=["nx", "ny"])
+
+        # get nlon/nlat for input/output datsets
+        nlon_i = ds_in.sizes["yh"]
+        nlat_i = ds_in.sizes["xh"]
+        nlon_o = len(self.lons1d_out)
+        nlat_o = len(self.lats1d_out)
+        self.ires = f"{nlon_i}x{nlat_i}"
+        self.ores = f"{nlon_o}x{nlat_o}"
+
+        # paths to interpolation weights files
+        if "weights_file_t2t" in self.config.keys():
+            weights_file_t2t = self.config["weights_file_t2t"]
+            weights_file_u2t = self.config["weights_file_u2t"]
+            weights_file_v2t = self.config["weights_file_v2t"]
+        if weights_file_t2t is None:
+            weights_file_t2t = f"weights-{self.ires}.Ct.{self.ores}.Ct.{self.interp_method}.nc"
+            weights_file_u2t = f"weights-{self.ires}.Cu.{self.ires}.Ct.{self.interp_method}.nc"
+            weights_file_v2t = f"weights-{self.ires}.Cv.{self.ires}.Ct.{self.interp_method}.nc"
+
+        # create regridding instances
+        reuse = os.path.exists(weights_file_t2t)
+        self.rg_tt = xe.Regridder(
+            ds_in_t,
+            grid_out,
+            self.interp_method,
+            periodic=True,
+            reuse_weights=reuse,
+            filename=weights_file_t2t,
+        )
+        reuse = os.path.exists(weights_file_u2t)
+        self.rg_ut = xe.Regridder(
+            ds_in_u,
+            ds_in_t,
+            self.interp_method,
+            periodic=True,
+            reuse_weights=reuse,
+            filename=weights_file_u2t,
+        )
+        reuse = os.path.exists(weights_file_v2t)
+        self.rg_vt = xe.Regridder(
+            ds_in_v,
+            ds_in_t,
+            self.interp_method,
+            periodic=True,
+            reuse_weights=reuse,
+            filename=weights_file_v2t,
+        )
+
+    def regrid(self, ds_in: xr.Dataset) -> xr.Dataset:
+        """
+        Regrid a single ocean file.
+        :param ds_in: Input MOM6 xarray Dataset to regrid.
+        :return: Regridded MOM6 xarray Dataset
+        """
+        ds_out = []
+
+        for var in list(ds_in.keys()):
+            coords = ds_in[var].coords.to_index()
+
+            # must have 3 coordinates, otherwise append copy
+            if len(coords.names) <= 2:
+                ds_out.append(ds_in[var])
+                continue

```

Comment:
Currently all 23 variables in MOM6 dataset have time coordinate, and the original code is written with that assumption. I wanted to add "unregriddable" variables like `average_T1(time)`, `time_bnds(time,nv)` to the output dataset, so I added the code for copying a variable without regridding it. I have now changed it so that it can handle 2D vars without the time coordinate. It checks if a coordinate uses any two of ("yh","xh","yq","xq"), in which case it will regrid it otherwise it will append a copy. Let me know if this better.